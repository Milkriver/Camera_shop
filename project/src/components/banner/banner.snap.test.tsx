import {render} from '@testing-library/react';
import { createMemoryHistory } from 'history';
import HistoryRouter from '../history-router/history-router';
import Banner from './banner';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';


const history = createMemoryHistory();

describe('Component: Banner', () => {
  it('should render correctly', () => {
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);

    const store = mockStore({
        OFFERS: {
            promoOffer: {}
        }
    });
    const {container} = render(
        <Provider store={store}>
            <HistoryRouter history={history}>
                <Banner />
            </HistoryRouter>
        </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
