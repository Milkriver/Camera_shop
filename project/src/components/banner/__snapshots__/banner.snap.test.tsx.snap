// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Banner should render correctly 1`] = `
<div>
  <div
    class="banner"
  >
    <picture>
      <source
        srcset="undefined, undefined"
        type="image/webp"
      />
      <img
        alt="баннер"
        height="280"
        width="1280"
      />
    </picture>
    <p
      class="banner__info"
    >
      <span
        class="banner__message"
      >
        Новинка!
      </span>
      <span
        class="title title--h1"
      />
      <span
        class="banner__text"
      >
        Профессиональная камера от известного производителя
      </span>
      <a
        class="btn"
        href="/product/undefined"
      >
        Подробнее
      </a>
    </p>
  </div>
</div>
`;
